demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -0,0 +1,1035 @@
+
+    xResult = AzureIoTJSONWriter_AppendPropertyWithDoubleValue( &xWriter, sampleazureiotCOMMAND_AV_TEMP, strlen( sampleazureiotCOMMAND_AV_TEMP ), xDeviceAverageTemperature, sampleazureiotDOUBLE_DECIMAL_PLACE_DIGITS );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTJSONWriter_AppendPropertyWithStringValue( &xWriter, sampleazureiotCOMMAND_START_TIME, strlen( sampleazureiotCOMMAND_START_TIME ), ucCommandStartTimeValueBuffer, ulSinceTimeLength );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTJSONWriter_AppendPropertyWithStringValue( &xWriter, sampleazureiotCOMMAND_END_TIME, strlen( sampleazureiotCOMMAND_END_TIME ), ucCommandEndTimeValueBuffer, xEndTimeLength );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTJSONWriter_AppendEndObject( &xWriter );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    return AzureIoTJSONWriter_GetBytesUsed( &xWriter );
+}
+
+/**
+ * @brief Cloud message callback handler
+ */
+static void prvHandleCloudMessage( AzureIoTHubClientCloudToDeviceMessageRequest_t * pxMessage,
+                                   void * pvContext )
+{
+    ( void ) pvContext;
+
+    LogInfo( ( "Cloud message payload : %.*s \r\n",
+               pxMessage->ulPayloadLength,
+               pxMessage->pvMessagePayload ) );
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Direct command message callback handler
+ */
+static void prvHandleCommand( AzureIoTHubClientCommandRequest_t * pxMessage,
+                              void * pvContext )
+{
+    LogInfo( ( "Command payload : %.*s \r\n",
+               pxMessage->ulPayloadLength,
+               pxMessage->pvMessagePayload ) );
+
+    int32_t ulCommandPayloadLength;
+
+    AzureIoTHubClient_t * xHandle = ( AzureIoTHubClient_t * ) pvContext;
+
+    if( strncmp( sampleazureiotCOMMAND_MAX_MIN_REPORT, pxMessage->pucCommandName, strlen( sampleazureiotCOMMAND_MAX_MIN_REPORT ) ) == 0 )
+    {
+        /* Is for max min report */
+        ulCommandPayloadLength = prvInvokeMaxMinCommand( pxMessage->pvMessagePayload, pxMessage->ulPayloadLength );
+
+        if( ulCommandPayloadLength > 0 )
+        {
+            if( AzureIoTHubClient_SendCommandResponse( xHandle, pxMessage, 200,
+                                                       ucCommandPayloadBuffer,
+                                                       ulCommandPayloadLength ) != eAzureIoTSuccess )
+            {
+                LogInfo( ( "Error sending command response" ) );
+            }
+        }
+        else
+        {
+            LogInfo( ( "Error generating command payload" ) );
+        }
+    }
+    else
+    {
+        /* Not for max min report (not for this device) */
+        LogInfo( ( "Received command is not for this device" ) );
+
+        if( AzureIoTHubClient_SendCommandResponse( xHandle, pxMessage, 404,
+                                                   sampleazureiotCOMMAND_EMPTY_PAYLOAD,
```

Comment:
>sampleazureiotCOMMAND_EMPTY_PAYLOAD [](http://example.com/codeflow?start=51&length=35)

we can pass NULL

Reply:
This mimics the embedded so unless we change it there I'm going to leave as is. I think the payload is required to be JSON.